tests/providers/microsoft/azure/hooks/test_azure_synapse.py

```diff
@@ -0,0 +1,186 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import json
+from typing import Type
+from unittest.mock import MagicMock, patch
+
+import pytest
+from azure.identity import ClientSecretCredential, DefaultAzureCredential
+from azure.synapse.spark import SparkClient
+from pytest import fixture
+
+from airflow.models.connection import Connection
+from airflow.providers.microsoft.azure.hooks.synapse import AzureSynapseHook, AzureSynapseSparkBatchRunStatus
+from airflow.utils import db
+
+DEFAULT_SPARK_POOL = "defaultSparkPool"
+
+DEFAULT_SYNAPSE = "defaultSynapse"
+SYNAPSE = "testSynapse"
+
+DEFAULT_CONNECTION_CLIENT_SECRET = "azure_synapse_test_client_secret"
+DEFAULT_CONNECTION_DEFAULT_CREDENTIAL = "azure_synapse_test_default_credential"
+
+MODEL = object()
+NAME = "testName"
+ID = "testId"
+JOB_ID = 1
+
+
+def setup_module():
+    connection_client_secret = Connection(
+        conn_id=DEFAULT_CONNECTION_CLIENT_SECRET,
+        conn_type="azure_synapse",
+        host="https://testsynapse.dev.azuresynapse.net",
+        login="clientId",
+        password="clientSecret",
+        extra=json.dumps(
+            {
+                "extra__azure_synapse__tenantId": "tenantId",
+                "extra__azure_synapse__subscriptionId": "subscriptionId",
+                "extra__azure_synapse__spark_pool": DEFAULT_SPARK_POOL,
```

Comment:
This connection param is missing from the hook (it's not an extra in the connection anyway) and custom connection spec. Should be added to the hook or removed from the connection here?

Reply:
it should be removed from the connection here. It was  leftover from a previous implementation. I made the changes.